packages/blockchain/src/blockchain.ts

```diff
@@ -309,11 +309,14 @@ export class Blockchain implements BlockchainInterface {
    *
    * @param name - Optional name of the iterator head (default: 'vm')
    */
-  async getIteratorHead(name = 'vm'): Promise<Block> {
-    return this.runWithLock<Block>(async () => {
+  async getIteratorHead(name = 'vm'): Promise<Block | Buffer> {
+    return this.runWithLock<Block | Buffer>(async () => {
       // if the head is not found return the genesis hash
       const hash = this._heads[name] ?? this.genesisBlock.hash()
       const block = await this.getBlock(hash)
+      if (block === null) {
```

Comment:
Is it possible to reach this code block?  I traced down through the subsequent calls in `getBlock` and it looks like we don't have any places where the genesis block isn't instantiated as an actual `block`.  I tried contriving a test to call `getIteratorHead` without calling `blockchain._init` (where the genesis block gets created) and the `genesisBlock` is always in the db so the initial call to `getBlock` in line 316 will always return an actual `block`.
Here's my contrived test.  If I comment out the call to `_init`, `getIteratorHead` throws with an error complaining I haven't run `_init`.

```
tape('check blockchain constructor', async (t) => {
  const blockchain = new (Blockchain as any)()
  await blockchain._init()
  console.log(await blockchain.getIteratorHead())
  t.end()
})
```

All that to say, I'm wondering if we really need to have the return type of `getIteratorHead` be `<Block | Buffer>` since I don't think there's a place where this call to `getBlock` would actually return `null`.

Reply:
I wonder the same thing.

Where does the "return the genesis hash if the head not found" rule come from?

Reply:
Not sure, but I think we can safely remove that code block and revert the interface change there. Are there any other breaking changes?

Reply:
Not sure, but I think we can safely remove that code block and revert the interface change there. Are there any other breaking - changes?

Reply:
I don't think so.

I'm about to push a commit that rolls back that change and the downstream effects.
Seems like everything is still fine